sdk/formrecognizer/Azure.AI.FormRecognizer/src/ClientCommon.cs

```diff
@@ -63,7 +63,9 @@ public static RequestFailedException CreateExceptionForFailedOperation(ClientDia
                 index++;
             }
 
-            return diagnostics.CreateRequestFailedException(response, errorMessage, errorCode, errorInfo);
+            return async
+                ? await diagnostics.CreateRequestFailedExceptionAsync(response, errorMessage, errorCode, errorInfo).ConfigureAwait(false)
```

Comment:
ohh so we do need to keep the consistency here too? we can't just call one or the other?

Reply:
You can create a universal `CreateRequestFailedExceptionAsync` method that takes `bool` parameter (`ClientDiagnostics` is an internal class), but it will require Pavel's approval.

Reply:
Or you can wrap this call. Let me update the PR.